USER_MANUAL.md

```diff
@@ -1,12 +1,355 @@
+
+K sentences are then organized into one or *modules* which are stored in one or
+more *files*. In this scheme, files may *require* other files and modules may
+*import* other modules, giving rise to a hierarchy of files and modules. We
+give an intuitive sketch of the two levels of grouping in the diagram below:
+
+```k
+   example.k file
+  +=======================+
+  | requires ".." --------|--> File_1
+  | ...                   |
+  | requires ".." --------|--> File_N
+  |                       |
+  |  +-----------------+  |
+  |  | module ..       |  |
+  |  |   imports .. ---|--|--> Module_1
+  |  |   ...           |  |
+  |  |   imports .. ---|--|--> Module_M
+  |  |                 |  |
+  |  |   sentence_1    |  |
+  |  |   ...           |  |
+  |  |   sentence_K    |  |
+  |  | endmodule       |  |
+  |  +-----------------+  |
+  |                       |
+  +=======================+
+```
+
+where:
+
+-   files and modules are denoted by double-bordered and single-borded boxes
+    respectively;
+-   file or module identifiers are denoted by double dots (`..`);
+-   potential repititions are denoted by triple dots (`...`).
+
+In the end, we require that the file and module hierarchies both form a
+directed acyclic graph (DAG). This is, no file may recursively require itself,
+and likewise, no module may recursively import itself.
+
+We now zoom in further to discuss the various kinds of sentences contained in K
+specifications:
+
+1.  sentences that define our *system's primitives*, including:
+
+    -   **sort declarations:** define new categories of primitive datatypes
+    -   **Backus-Naur Form (BNF) grammar declarations:** define the
+        operators that inhabit our primitive datatypes
+    -   **lexical syntax declarations:** define lexemes/tokens for the
+        lexer/tokenizer
+    -   **syntax associativity declarations:** specify the
+        associativity/grouping of our declared operators
+    -   **syntax priority declarations:** specify the priority of
+        potential ambiguous operators
+
+2.  sentences that define our *system's state*, including:
+
+    -   **configuration declarations:** define labelled, hierarchical records
+        using an nested XML-like syntax
+
+3.  sentences that define our *system's behavior*, including:
+
+    -   **context declarations:** describe how primitives and configurations
+        can simplify
+    -   **context alias declarations:** define templates that can generate new
+        contexts
+    -   **rule declarations:** define how the system transitions from one state
+        to the next
+
+### K Syntax Overview
+
+Here we provide an overview of the top-level K syntax using a K-style BNF
+notation. Note that this is not a full definition; we focus instead on the
+overall structure of the specification. Syntactic categories which are not
+defined are denoted by names surrounded by curly braces (`{}`). In particular,
```

Comment:
if we're going to be giving a grammar, we shouldn't make it incomplete like this. It's possible to define the syntax of all of these identifiers and such fairly readily.

Reply:
I removed this section from this PR because it needs more polish.